File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/command/simple/ChangeWeekendModeCommand.java
package com.happs.ximand.ringcontrol.model.object.command.simple;

public class ChangeWeekendModeCommand extends SimpleBluetoothCommand {

    public static final byte COMMAND_CODE = 21;

    public ChangeWeekendModeCommand(byte value) {
        super(COMMAND_CODE, value);
        if (value > 2) {
            throw new IllegalArgumentException();
        }
    }

}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/command/simple/ChangeWeekendModeCommand.kt
package com.happs.ximand.ringcontrol.model.`object`.command.simple

class ChangeWeekendModeCommand(value: Byte) : SimpleBluetoothCommand(CODE, value) {

    companion object {
        const val CODE: Byte = 21
    }

    init {
        require(value <= 2)
    }
}